File: app/src/main/java/com/dcurreli/spese/adapters/ListaSpeseAdapter.java
package com.dcurreli.spese.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.core.view.GravityCompat;
import androidx.navigation.NavController;
import androidx.recyclerview.widget.RecyclerView;

import com.dcurreli.spese.R;
import com.dcurreli.spese.databinding.ActivityMainBinding;
import com.dcurreli.spese.databinding.HomeFragmentBinding;
import com.dcurreli.spese.objects.ListaSpese;
import com.dcurreli.spese.utils.GenericUtils;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class ListaSpeseAdapter extends RecyclerView.Adapter<ListaSpeseAdapter.MyViewHolder> {

    private Context context;
    private final ArrayList<ListaSpese> listaSpeseList;
    private final HomeFragmentBinding binding;
    private final NavController navController;

    public ListaSpeseAdapter(Context context, ArrayList<ListaSpese> listaSpeseList, HomeFragmentBinding binding, NavController navController) {
        this.context = context;
        this.listaSpeseList = listaSpeseList;
        this.binding = binding;
        this.navController = navController;
    }
    public Context getContext() {
        return context;
    }

    public void setContext(Context context) {
        this.context = context;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(context).inflate(R.layout.home_fragment_lista_spese, parent, false);
        return new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        ListaSpese listaSpese = listaSpeseList.get(position);
        holder.nomeSpesa.setText(listaSpese.getNome());

        //Gestisco l'evento on click
        holder.relativeLayout.setOnClickListener(view -> {
            //Navigo sul fragment successivo passandogli il bundle con id lista e nome lista
            navController.navigate(R.id.action_HomeFragment_to_loadSpeseFragment, GenericUtils.INSTANCE.createBundleForListaSpese(listaSpese.getId(), holder.nomeSpesa.getText().toString()));
        });

        if(listaSpese.isSaldato()){
            holder.saldato.setBackgroundColor(ContextCompat.getColor(context, android.R.color.holo_green_dark));
        }else{
            holder.saldato.setBackgroundColor(ContextCompat.getColor(context, android.R.color.holo_red_dark));
        }
    }

    @Override
    public int getItemCount() {
        return listaSpeseList.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {

        TextView nomeSpesa;
        ImageView saldato;
        RelativeLayout relativeLayout;

        public MyViewHolder(@NotNull View itemView) {
            super(itemView);
            nomeSpesa = itemView.findViewById(R.id.nome_spesa);
            relativeLayout = itemView.findViewById(R.id.spesa_list_first_layout);
            saldato = itemView.findViewById(R.id.saldato_bar);
        }
    }

}

File: app/src/main/java/com/dcurreli/spese/adapters/ListaSpeseAdapter.kt
package com.dcurreli.spese.adapters

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.navigation.NavController
import androidx.recyclerview.widget.RecyclerView
import com.dcurreli.spese.R
import com.dcurreli.spese.data.entity.ExpensesList
import com.dcurreli.spese.utils.GenericUtils

class ListaSpeseAdapter(private val navController: NavController) : RecyclerView.Adapter<ListaSpeseAdapter.ViewHolder>() {

    private var expensesListList = ArrayList<ExpensesList>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.home_fragment_lista_spese, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        val expensesList = expensesListList[position]
        holder.nomeSpesa.text = expensesList.name

        //Gestisco l'evento on click
        holder.relativeLayout.setOnClickListener {
            //Navigo sul fragment successivo passandogli il bundle con id lista e nome lista
            navController.navigate(
                R.id.action_HomeFragment_to_loadSpeseFragment,
                GenericUtils.createBundleForListaSpese(
                    expensesList.id!!,
                    holder.nomeSpesa.text.toString()
                )
            )
        }

        holder.relativeLayout.setBackgroundResource(if (expensesList.paid) R.drawable.lista_liste_saldato else R.drawable.lista_liste_da_saldare)

    }

    override fun getItemCount(): Int {
        return expensesListList.size
    }

    @SuppressLint("NotifyDataSetChanged")
    fun addItems(newExpensesListList : List<ExpensesList>){
        expensesListList.clear()
        expensesListList.addAll(newExpensesListList)
        notifyDataSetChanged()
    }

    // Holds the views for adding it to image and text
    class ViewHolder(ItemView: View) : RecyclerView.ViewHolder(ItemView) {
        val nomeSpesa: TextView = itemView.findViewById(R.id.nome_spesa)
        val relativeLayout: RelativeLayout = itemView.findViewById(R.id.spesa_list_layout)
    }
}